owltorrent-core/src/test/java/edu/rice/owltorrent/core/serialization/TempParserTest.java

```diff
@@ -0,0 +1,51 @@
+package edu.rice.owltorrent.core.serialization;
+
+import edu.rice.owltorrent.common.entity.Torrent;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+public class TempParserTest {
+    static String coreRoot = System.getProperty("user.dir");
+    static String torrentFileLocation = coreRoot + "/CourseraDSFundamentals.torrent";
+    static File file = new File(torrentFileLocation);
+
+    @Test
+    public void parse() throws IOException {
+        Torrent torrent = TempParser.parse(file);
+
+        assertEquals("udp://tracker.leechers-paradise.org:6969/announce", torrent.getAnnounceURL());
+        assertEquals("Data Science Fundamentals with Python and SQL", torrent.getName());
+        assertEquals(1464, torrent.getPieces().size());
+        assertEquals(524288, torrent.getPieceLength());
+        assertEquals(807986536, torrent.getLength());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void parse_nullInput() throws IOException {
+        TempParser.parse(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void bencode_nullInput() throws IOException {
+        TempParser.bencode(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void extractAttributes_nullInput() {
+        TempParser.extractAttributes(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void extractPieces_nullInput() {
+        TempParser.extractPieces(null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void getFileLength_nullInput() {
```

Comment:
This test doesn't actually verify anything. Maybe you forgot the nonNull annotation on getFileLength?

Reply:
Fixed in revision